src/YMenuButton.cc

```diff
@@ -264,3 +264,36 @@ YMenuButton::getProperty( const string & propertyName )
 	return YWidget::getProperty( propertyName );
     }
 }
+
+YMenuItem *
+YMenuButton::findItem( std::vector<std::string> & path ) const
+{
+    return findItem( path.begin(), path.end(), itemsBegin(), itemsEnd());
+}
+
+YMenuItem *
+YMenuButton::findItem( std::vector<std::string>::iterator path_begin,
+                       std::vector<std::string>::iterator path_end,
+                       YItemConstIterator begin,
+                       YItemConstIterator end ) const
+{
+    for ( YItemConstIterator it = begin; it != end; ++it )
+    {
+        YMenuItem * item = dynamic_cast<YMenuItem *>(*it);
+        // Test that dynamic_cast didn't fail
+        if ( !item ) return nullptr;
+
+        if( item->label() == *path_begin ) {
+            if ( std::next(path_begin) == path_end ) {
+                // Only return items which can trigger action, intermediate items only open nested popup, so continue looking
+                if(item->hasChildren()) continue;
```

Comment:
Coding style; please do

```C++
if ( item->hasChildren() )
    continue;
```

Having a one-line `if` clause's statement on a separate line means you see in a debugger if it's executed or not. You don't see that when it's on the same line.

Other code locations here accordingly.

Reply:
Fixed, thanks!